tests/test_openjdk.py

```diff
@@ -152,3 +172,66 @@ def test_jdk_extended(
 
     for check in params.expected_err_strings:
         assert check in testout.stderr
+
+
+@pytest.mark.parametrize(
+    "container_per_test",
+    CONTAINER_IMAGES_CASSANDRA,
+    indirect=["container_per_test"],
+)
+def test_jdk_cassandra(container_per_test):
+    """Starts the Cassandra DB and executes some write and read tests
+    using the cassandra-stress
+    """
+
+    cassandra_versions = container_per_test.connection.check_output(
+        "git ls-remote --tags https://gitbox.apache.org/repos/asf/cassandra.git"
+    )
+
+    cassandra_version = Version(0, 0, 0)
+    for line in cassandra_versions.splitlines():
+        match = re.search(r"cassandra-(\d).(\d).(\d)$", line)
+        if match:
+            cur_ver = Version(
+                int(match.group(1)), int(match.group(2)), int(match.group(3))
+            )
+            if cur_ver > cassandra_version:
+                cassandra_version = cur_ver
+
+    container_per_test.connection.run_expect(
+        [0],
+        f"curl -OL https://downloads.apache.org/cassandra/{cassandra_version}/apache-cassandra-{cassandra_version}-bin.tar.gz",
+    )
+
+    container_per_test.connection.run_expect(
+        [0], f"tar xzvf apache-cassandra-{cassandra_version}-bin.tar.gz"
+    )
+
+    container_per_test.connection.run_expect(
+        [0],
+        f"export JAVA_HOME=/usr/ && cd apache-cassandra-{cassandra_version}/ && bin/cassandra -R >/var/log/cassandra.log",
+    )
+
+    found = False
+    for t in range(40):
+        time.sleep(10)
+        if (
+            "state jump to NORMAL"
+            in container_per_test.connection.check_output(
+                "cat /var/log/cassandra.log"
+            )
+        ):
```

Comment:
nitpick: https://testinfra.readthedocs.io/en/latest/modules.html?highlight=file#testinfra.modules.file.File.contains is a better fit for this

Reply:
done